app/controllers/application_controller.rb

```diff
@@ -91,6 +91,15 @@ def set_current_location_name
     cookies['location'] = params[:location] unless params[:location] == nil
   end
 
+  def check_for_empty_password
+    if session[:original_user] and session[:"warden.user.user.key"]
```

Comment:
Why was the warden part required? I'd imagine you'd be able to do this with just the original_user part.